server/report/extractor.py

```diff
@@ -0,0 +1,186 @@
+def _numeral(value):
+    """
+    Change to integer/float
+    """
+    try:
+        return int(value)
+    except ValueError:
+        return float(value)
+
+
+def _percentage(value):
+    """
+    Change percentage text to integer/float
+    """
+    return _numeral(value.replace(' ', '').replace('%', ''))
+
+
+def generate_hierarchy(obj, data):
+    result = {}
+
+    result['name'] = obj['name']
+    if obj.get('size'):
+        result['size'] = _numeral(data[obj['size']])
+
+    childrens = obj.get('children')
+    if childrens and len(childrens):
```

Comment:
Just `if childens:` is sufficient. And, children is plural of child. :P

Reply:
Fixed